core/sail/shacl/src/test/resources/shaclDatatypeSparqlTarget.ttl

```diff
@@ -0,0 +1,34 @@
+@base <http://example.com/ns> .
+@prefix ex: <http://example.com/ns#> .
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix foaf: <http://xmlns.com/foaf/0.1/>.
+
+ex:PersonShape
+	a sh:NodeShape  ;
+		sh:target [
+                		a sh:SPARQLTarget ;
+                		sh:select """
+                		        PREFIX ex: <http://example.com/ns#>
+                                        PREFIX owl: <http://www.w3.org/2002/07/owl#>
+                                        PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+                                        PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+                                        PREFIX sh: <http://www.w3.org/ns/shacl#>
+                                        PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+
+                			SELECT ?this
+                			WHERE {
+                				?this a rdfs:Resource .
+                			}
+                			""" ;
+                ] ;
+
+	sh:property [
+		sh:path foaf:age ;
+		sh:datatype xsd:int ;
+
+	] .
+
```

Comment:
Example of the SPARQL target support.

Note: You have to include prefixes in the string. We don't support `sh:prefixes`!